nilearn/input_data/nifti_spheres_masker.py

```diff
@@ -0,0 +1,232 @@
+"""
+Transformer for computing seeds signals.
+"""
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn import neighbors
+from sklearn.externals.joblib import Memory
+
+from .. import _utils
+from .._utils import logger, CacheMixin
+from .._utils.niimg_conversions import check_niimg, check_niimg_3d
+from .. import signal
+from .. import image
+from .. import masking
+
+
+def _iter_signals_from_spheres(seeds, niimg, radius, mask_img=None):
+    seeds = list(seeds)
+    niimg = check_niimg(niimg)
+    affine = niimg.get_affine()
+
+    # Compute world coordinates of all in-mask voxels.
+
+    if mask_img is not None:
+        mask_img = check_niimg_3d(mask_img)
+        mask_img = image.resample_img(mask_img, target_affine=affine,
+                                      target_shape=niimg.shape[:3],
+                                      interpolation='nearest')
+        mask, _ = masking._load_mask_img(mask_img)
+        mask_coords = list(np.where(mask != 0))
+
+        X = masking._apply_mask_fmri(niimg, mask_img)
+    else:
+        mask_coords = list(zip(*np.ndindex(niimg.shape[:3])))
+        X = niimg.get_data().reshape([-1, niimg.shape[3]]).T
+    mask_coords.append(np.ones(len(mask_coords[0]), dtype=np.int))
+    mask_coords = np.asarray(mask_coords)
+    mask_coords = np.dot(affine, mask_coords)[:3].T
+
+    if radius is not None:
+        # Fix for early version of sklearn (not sure)
+        radius += 1e-6
+
+    clf = neighbors.NearestNeighbors(radius=radius)
+    A = clf.fit(mask_coords).radius_neighbors_graph(seeds)
```

Comment:
Isn't there any moe readbale way to perform nearest neighbor assignment (maybe not, I'm just asking) ?